lib/BackgroundJobs/IndexerJob.php

```diff
@@ -111,13 +113,22 @@ protected function getBatchSize(): int {
 		return $this->appConfig->getAppValueInt('indexing_batch_size', 100);
 	}
 
+	protected function getMaxIndexingTime(): int {
+		return $this->appConfig->getAppValueInt('indexing_max_time', self::DEFAULT_MAX_INDEXING_TIME);
+	}
+
 	/**
 	 * @param QueueFile[] $files
 	 * @return void
 	 * @throws \RuntimeException|\ErrorException
 	 */
 	protected function index(array $files): void {
+		$maxTime = $this->getMaxIndexingTime();
+		$startTime = time();
 		foreach ($files as $queueFile) {
+			if ($startTime + $maxTime > time()) {
+				break;
+			}
```

Comment:
I may be wrong but it looks like the loop breaks on the first iteration. Something like this would be better: `if (time() - $startTime > $maxTime)` or just flip the comparision.

Reply:
Aah, you're right. Looks like my brain was still in vacation mode :D
```suggestion
			if ($startTime + $maxTime < time()) {
				break;
			}
```